src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -163,14 +168,62 @@ public KeyManager[] getKeyManagers(URI privateKeyUri, URI certificateUri)
         return provider.getKeyManagers(privateKeyUri, certificateUri);
     }
 
+    /**
+     * Get certificate chain using private and certificate URIs.
+     *
+     * @param privateKeyUri  private key URI
+     * @param certificateUri certificate URI
+     * @return X509Certificate list that corresponds to the given private key and certificate URIs
+     * @throws ServiceUnavailableException if crypto key provider service is unavailable
+     * @throws KeyLoadingException         if crypto key provider service fails to load key
+     * @throws CertificateException if a valid certificate is not found at the URI specified
+     * @throws IOException         if reading file at the specified URI fails
+     */
+    public List<X509Certificate> getCertificateChain(URI privateKeyUri, URI certificateUri)
+            throws ServiceUnavailableException, KeyLoadingException, IOException, CertificateException {
+        logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                .log("Getting the certificate chain using private key and certificate URIs");
+        CryptoKeySpi provider = selectCryptoKeyProvider(privateKeyUri);
+        try {
+            return provider.getCertificateChain(privateKeyUri, certificateUri);
+        } catch (UnsupportedOperationException e) {
+            logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                    .log("Could not find the certificate using the provider implementation."
+                            + " Getting it from the key manager.");
+            KeyManager[] km = provider.getKeyManagers(privateKeyUri, certificateUri);
+            if (km.length != 1 || !(km[0] instanceof X509KeyManager)) {
+                logger.atError().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                        .log("Unable to find the key manager with the given URIs.");
+                return null;
```

Comment:
Could we return an empty list instead? That ensures we adhere to the signature we are telling clients we adhere to

```suggestion
                return Collections.< X509Certificate>emptyList();
```